multi-node/aws/pkg/config/config.go

```diff
@@ -385,26 +426,78 @@ func (cfg Cluster) valid() error {
 		return fmt.Errorf("invalid vpcCIDR: %v", err)
 	}
 
-	_, instancesNet, err := net.ParseCIDR(cfg.InstanceCIDR)
-	if err != nil {
-		return fmt.Errorf("invalid instanceCIDR: %v", err)
-	}
-	if !vpcNet.Contains(instancesNet.IP) {
-		return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s)",
-			cfg.VPCCIDR,
-			cfg.InstanceCIDR,
-		)
-	}
-
 	controllerIPAddr := net.ParseIP(cfg.ControllerIP)
 	if controllerIPAddr == nil {
 		return fmt.Errorf("invalid controllerIP: %s", cfg.ControllerIP)
 	}
-	if !instancesNet.Contains(controllerIPAddr) {
-		return fmt.Errorf("instanceCIDR (%s) does not contain controllerIP (%s)",
-			cfg.InstanceCIDR,
-			cfg.ControllerIP,
-		)
+
+	if len(cfg.Subnets) == 0 {
```

Comment:
@colhom Thanks for your question!

I guess doing that([overwriting `InstanceCIDR`](https://github.com/mumoshu/coreos-kubernetes/blob/multi-az-workers/multi-node/aws/pkg/config/config.go#L78) with a default value and determining [implicit `Subnets`](https://github.com/mumoshu/coreos-kubernetes/blob/multi-az-workers/multi-node/aws/pkg/config/config.go#L86) from the top-level `InstanceCIDR` and `AvailabilityZone`) before validation prevents us doing the validation: `Did the user specified both top-level InstanceCIDR/AvailabilityZone and Subnets?` defined [here](https://github.com/mumoshu/coreos-kubernetes/blob/multi-az-workers/multi-node/aws/pkg/config/config.go#L455) with the [test](https://github.com/mumoshu/coreos-kubernetes/blob/multi-az-workers/multi-node/aws/pkg/config/config_test.go#L415)

It's a bit confusing but, thinking more generally, I took `c.valid()` as a validation of an user input. To structurally validate the user input, I thought it would be better not to structurally touch it before validation.

Does this make sense to you?

Btw, the order of `valid()` call and backward-compatiblity logic, defaults are not very intuitive in my code and I wish I could somehow make it more intuitive, too 👍
Just not coming up with a nicer idea though 😢